### ISSUE TITLE
pre-training weight

### ISSUE THREAD

**Issue by Ocean728** [yaorong0921/Driver-Intention-Prediction#1]

**Comment by yaorong0921** [comment]
Hi,
the link to the drive now is updated and you may download our trained weights from the drive.
If you still cannot have the access to it, pls let me know.

Best,

**Comment by Ocean728** [comment]
Hi,
I'm so glad to have received your response.
the  trained weights from the drive had been downloaded sucessfully.
I cannot find a test file to evaluate 3D-ResNet50 in the repo. and how can I finish this work with "save_best_3DResNet50.pth".
Looking forward to your reply.

Sincerely,

**Comment by yaorong0921** [comment]
Hi,
to evaluate it the trained model, you can add two args in the 
"run-3DResnet.sh":
--resume_path  /path_to_model/save_best_3DResNet50.pth
--no_train

I hope it helps you.

Best,

**Comment by Ocean728** [comment]
thank you very much.
I would like to consult two questions:
1, there is a SyntaxError to the line 1 of the file "run-3DResnet.sh "  when I run it, and the error point to python.
2, There seems to be no file to test a single video in this repo，so how to show the performance of the framework？
thank you for you time and patience.

**Comment by yaorong0921** [comment]
Hi,
1. could you show your error?
2. By performance, do you mean to give a single video and then show the predicted action? 
Once you have trained two branches for in- and outside videos, you can train the classifier using the features from two branches.
Concretely: 
The features for inside videos F_I are the 2048-dim features before the last fully-connected layer in 3D Resnet.
The features for outside F_O videos are the (32*112*176) output tensors of the last layer of ConvLSTM Encoder.

Once you have the features, you can use the architecture list in TABLE II in the paper to train the classifier: F_O is first put into four Conv-Blocks and then concatenated together with F_I to form F_both. F_both should be a 3072-dim feature. F_both is fed into two fully-connected layers used for making the final decision.

Best,

**Comment by Ocean728** [comment]
The error is shown in the image below.
![erorr](https://user-images.githubusercontent.com/73243594/112707904-b3ccce00-8ee9-11eb-9065-02cc1cfcb2f3.PNG)

**Comment by yaorong0921** [comment]
Hi, 
please add a new file named "opts.json" and try to run the code.

Best,

**Comment by Ocean728** [comment]
Thank you for your explanation！The&nbsp; path is not configured for the previous error.However，there is a new error as following.






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "yaorong0921/Driver-Intention-Prediction"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年3月30日(星期二) 晚上6:14
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [yaorong0921/Driver-Intention-Prediction] pre-training weight (#1)





 
Hi,
 please add a new file named "opts.json" and try to run the code.
 
Best,

**Comment by Ocean728** [comment]
Thank you for your explanation！
The file path is not configured correctly for the previous error.
However，there is a new error as following.
![error](https://user-images.githubusercontent.com/73243594/113080507-9e1d1880-9209-11eb-9439-08a97fd8a67e.png)

**Comment by Ocean728** [comment]
there is a error as following when I evaluate this model with the "save_best_3DResNet50.pth".
It is necessary to inform that because the dataset is very large, I only use part of the data in the “face_camera” to evaluate this model.
Thank you for your explanation.
![error](https://user-images.githubusercontent.com/73243594/113271735-6a2b1b80-930d-11eb-97cd-047adec72bdd.jpg)

**Comment by yaorong0921** [comment]
Hallo,

did you use your own annotation files (.csv) files?
You could check whether the video_names[i] provides  the item like "
rchange/20141220_162850_1165_1315/".

Best,

Am Do., 1. Apr. 2021 um 11:13 Uhr schrieb ROBERT ***@***.***>:

**Comment by Ocean728** [comment]
Hi,

I'm sorry I didn't describe the question clearly.
I plan to make an attempt with a part of the data first, and generate  my own files(.csv).
1. when I run the "extract_images.py" ,the item like"Output #0, image2, to './face_camera\lchange\20141019_091136_63_213\image-%4d.png'".And when I run the "n_fold_Brain4cars.py", the item like"lchange\20141019_091136_63_213\,lchange\20141019_091136_63_213,126,validation".
2. how can I get the features from two branches?
3. Is there a pre-training weight of ConLSTM here?

Best,

**Comment by yaorong0921** [comment]
Hi,

1. I think you generated the files in the windows system. Pls replace the
'\' by '/' and see whether it works.
2. Once you have trained two branches for in- and outside videos, you can
train the classifier using the features from two branches
Concretely:
The features for inside videos F_I are the 2048-dim features before the
last fully-connected layer in 3D Resnet.
The features for outside F_O videos are the (32*112*176) output tensors of
the last layer of ConvLSTM Encoder.

Once you have the features, you can use the architecture list in TABLE II
in the paper to train the classifier: F_O is first put into four
Conv-Blocks and then concatenated together with F_I to form F_both. F_both
should be a 3072-dim feature. F_both is fed into two fully-connected layers
used for making the final decision.

3. So there is no pretrained weight of ConvLSTM and I trained ConvLSTM
using the Optical Flow images from scratch.  I uploaded the ConvLSTM model
trained with fold0.csv to
https://bwstaff-my.sharepoint.com/:f:/g/personal/yao_rong_bwstaff_de/EpmuNb3eB7hPgv2DmeBrQ1ABqgQ6uInXudrpfQQyPgmJZA?e=6LkUKH
. You may try it but also train from scratch may be better since we use
different fold devisions.


Best,

Am Fr., 9. Apr. 2021 um 11:38 Uhr schrieb ROBERT ***@***.***>:

**Comment by Ocean728** [comment]
yes.I generated the files in the windows system.
I have changed the '\' to '/' by added the code like  image below
![-38fdcdbe20e19529](https://user-images.githubusercontent.com/73243594/114182536-5765a780-9908-11eb-9824-783ea36e9531.jpg)

and then I run the "run-3DResnet.sh"，but the program can not go on，like this image below. 
![6d89f4ec72300ca4](https://user-images.githubusercontent.com/73243594/114183266-1de16c00-9909-11eb-8e2d-b6f1bb1957c0.png)

**Comment by yaorong0921** [comment]
It seems that you did not read any video inputs. Pls check your
dataloader e.g. len(train_data) that you have input data.

Am Fr., 9. Apr. 2021 um 14:57 Uhr schrieb ROBERT ***@***.***>:

**Comment by Ocean728** [comment]
and also all tensors are zero when I train the model with pre-training weight，like this
![-5d46d9343b772c23](https://user-images.githubusercontent.com/73243594/114184328-52a1f300-990a-11eb-93b0-ddfb486ac472.png)

**Comment by yaorong0921** [comment]
If you load the pretrained model "resnet-50-kinetics", there will be a
message printed out. Pls be careful about all the path when you load in,
since os.path.join() behaves not equally on Windows


Am Fr., 9. Apr. 2021 um 15:06 Uhr schrieb ROBERT ***@***.***>:

**Comment by Ocean728** [comment]
Yes，the message of the pretrained model "resnet-50-kinetics" has been printed out. 
the error occurred not only at the '\'，but also at the comma separator when I run "n_fold_Brain4cars.py"，as shown in the following image. 
![1945b9ceadf68b1e](https://user-images.githubusercontent.com/73243594/114267531-d10d9c00-99c1-11eb-898f-33ccde450f75.png)

**Comment by yaorong0921** [comment]
This is because Glob() also returns the path with “/“, and to extract the
target label, the path needs to find “/“ first: (line 35 in
“n_fold_Braun4cars.py”.
So what you can try is to change line 35 to: = fbase.find(‘\’, 1)

ROBERT ***@***.***> schrieb am Sa. 10. Apr. 2021 um 13:03:

**Comment by Harlan728** [comment]
Thank you very much.

The cause of the error may be that the data is not loaded when  "run-3DResnet.sh " was run.
However, there were not any errors could be found in "opts.json" when I checked all paths.
Could you please help me find out what the possible problems are in this repo at your convenience.(The data of face_camera has been deleted in advance)
[Driver-Intention-Prediction.zip](https://github.com/yaorong0921/Driver-Intention-Prediction/files/6296841/Driver-Intention-Prediction.zip)

Esteem it a favor!

**Comment by Ocean728** [comment]
Thank you very much.

The cause of the error may be that the data is not loaded when "run-3DResnet.sh " was run.
However, there were not any errors could be found in "opts.json" when I checked all paths.
Could you please help me find out what the possible problems are in this repo at your convenience.(The data of face_camera has been deleted in advance)
[Driver-Intention-Prediction.zip](https://github.com/yaorong0921/Driver-Intention-Prediction/files/6296938/Driver-Intention-Prediction.zip)


Esteem it a favor!

**Comment by yaorong0921** [comment]
Hi,
I cheked the error and I thought it could be that the video_path is not
correct. In the image, the video_path is "E:/pycharm....
prediction\\brain4cars...". Again, this mixed up of '/' or '\' can cause
the problem that the dataloader does not load any data.
Please check line 116 in Brain4cars.py.
Also, please print out the len(dataloader), to make sure there is data
loaded.

Best,


Am Mo., 12. Apr. 2021 um 15:16 Uhr schrieb ROBERT ***@***.***

**Comment by Ocean728** [comment]
Thank you for your kind help.
I checked this repo again,and here are some supplementary descriptions. 
1. The video_path could be found,as shown below.( part of face_camera)
![1(1)](https://user-images.githubusercontent.com/73243594/115114613-ecdee800-9f55-11eb-86f2-9a4d7c72230d.PNG)

And the defined "make_dataset" is nomal,but  the dataloader still does not load any data.
2. Maybe,the error happened in the  "class Brain4cars_Inside". The"get_default_video_loader" seems to be invalid,and the three parameters,video_dir_path, frame_indices and image_loader in"def video_loader"cannot be found. 
The "self.data" could be printed out normally,as shown below. 
![3](https://user-images.githubusercontent.com/73243594/115114598-d2a50a00-9f55-11eb-9f0d-4b66450def28.PNG)

All thoes problems above are encountered when evaluating the "3D-ResNet50" with the "save_best_3DResNet50.pth" in the drive.
However,after changing "n_classes" from 400 to 5, the model can be trained with the "save_best_3DResNet50.pth" successfully.
So,I wonder why the "save_best_3DResNet50.pth" can be used for training but not directly for evaluation.

There are another two questions：
1. Why is the learning rate  0.0 
![qxlarge-dsc-D30BFBFA61F2554628C8DCB757400AB9](https://user-images.githubusercontent.com/73243594/115394810-cf558c80-a1a8-11eb-823c-22cb2d20796e.png)

2. There are only the last five epochs(--epochs 30) evaluated, When evaluating the model with "save_best.pth"  trained by myself.
![qxlarge-dsc-CD27E315251A05A54909ED070D7527BB](https://user-images.githubusercontent.com/73243594/115394868-da102180-a1a8-11eb-9d65-73143fb897ca.png)

**Comment by Sidd1609** [comment]
> Hi, the link to the drive now is updated and you may download our trained weights from the drive. If you still cannot have the access to it, pls let me know.
> 
> Best,

The link is not working as of 23rd march, 2023. Please could you share a pre-trained weight.

**Comment by Sidd1609** [comment]
How were you able to train the model using "save_best_3DResNet50.pth"? @Ocean728.

data loader is not working properly.

**Comment by Ocean728** [comment]
I don't.



---Original---
From: ***@***.***&gt;
Date: Tue, Mar 28, 2023 02:49 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [yaorong0921/Driver-Intention-Prediction] pre-training weight(#1)




 
How were you able to train the model using "save_best_3DResNet50.pth"? @Ocean728.
 
data loader is not working properly.
